x/ccv/parent/keeper/relay.go

```diff
@@ -112,23 +112,23 @@ func (k Keeper) OnTimeoutPacket(ctx sdk.Context, packet channeltypes.Packet, dat
 // in a packet over the CCV channel.
 func (k Keeper) EndBlockCallback(ctx sdk.Context) {
 	valUpdateID := k.GetValidatorSetUpdateId(ctx)
-	// k.IncrementValidatorSetUpdateId(ctx) TODO: move this here
+	k.IncrementValidatorSetUpdateId(ctx)
 	k.IterateBabyChains(ctx, func(ctx sdk.Context, chainID string) (stop bool) {
 		valUpdates := k.stakingKeeper.GetValidatorUpdates(ctx)
 		if len(valUpdates) != 0 {
+			fmt.Println(valUpdateID)
 			k.SendPacket(ctx, chainID, valUpdates, valUpdateID)
-			k.IncrementValidatorSetUpdateId(ctx) // TODO: this needs to be moved out of this scope
 			k.SetValsetUpdateBlockHeight(ctx, valUpdateID, uint64(ctx.BlockHeight()))
 		}
 		return false
 	})
 }
 
-// OnRecvPacket slahes and jails the given validator in the packet data
+// OnRecvPacket slashes and jails the given validator in the packet data
 func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data ccv.ValidatorDowntimePacketData) exported.Acknowledgement {
-	if childChannel, ok := k.GetChannelToChain(ctx, packet.DestinationChannel); !ok && childChannel != packet.DestinationChannel {
+	if chainID, ok := k.GetChannelToChain(ctx, packet.DestinationChannel); !ok {
 		ack := channeltypes.NewErrorAcknowledgement(
-			fmt.Sprintf("packet sent on a channel %s other than the established child channel %s", packet.DestinationChannel, childChannel),
+			fmt.Sprintf("chain ID doesn't exist for channel ID: %s", chainID),
```

Comment:
If `ChannelToChain` is not set for a particular channel, it doesn't mean that the chain doesn't exist, but rather that the channel was not established, i.e., https://github.com/cosmos/interchain-security/blob/fc59e104ef25203f92d411d7b293d7c9a980d711/x/ccv/parent/keeper/keeper.go#L243.